chain/types/legacy/event.go

```diff
@@ -0,0 +1,90 @@
+package legacy
```

Comment:
Legacy types cordoned off in a subpackage for easier identification and maintenance.